File: Listadecomprasdelsupermercado.py
# Lista de compras - Estructura dinámica
compras = []

print("=== LISTA DE COMPRAS DEL SUPERMERCADO ===")
print("Ingrese al menos 3 productos:\n")

# Ingreso mínimo de 3 datos
for i in range(3):
    producto = input(f"Producto {i+1}: ").strip()
    compras.append(producto)

# Mostrar datos almacenados
print("\nLista de compras actual:")
for idx, prod in enumerate(compras, 1):
    print(f"{idx}. {prod}")

# Operación 1: Agregar un producto
nuevo = input("\nIngrese un nuevo producto a agregar: ").strip()
compras.append(nuevo)
print("\nLista después de agregar el producto:")
for idx, prod in enumerate(compras, 1):
    print(f"{idx}. {prod}")

# Operación 2: Eliminar un producto por índice
try:
    elim = int(input("\nIngrese el número del producto que desea eliminar: ")) - 1
    if 0 <= elim < len(compras):
        eliminado = compras.pop(elim)
        print(f"\nProducto eliminado: {eliminado}")
    else:
        print("Índice fuera de rango.")
except ValueError:
    print("Debe ingresar un número válido.")

# Lista final
print("\nLista final de compras:")
for idx, prod in enumerate(compras, 1):
    print(f"{idx}. {prod}")

File: ListaDeComprasDelSupermercado.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int main() {
    vector<string> compras;
    string producto;

    cout << "=== LISTA DE COMPRAS DEL SUPERMERCADO ===" << endl;
    cout << "Ingrese al menos 3 productos:\n" << endl;

    // Ingreso mínimo de 3 datos
    for(int i = 0; i < 3; i++) {
        cout << "Producto " << (i+1) << ": ";
        cin >> producto;
        compras.push_back(producto);
    }

    // Mostrar datos almacenados
    cout << "\nLista de compras actual:" << endl;
    for(size_t i = 0; i < compras.size(); i++) {
        cout << (i+1) << ". " << compras[i] << endl;
    }

    // Operación 1: Agregar
    cout << "\nIngrese un nuevo producto a agregar: ";
    cin >> producto;
    compras.push_back(producto);

    cout << "\nLista después de agregar el producto:" << endl;
    for(size_t i = 0; i < compras.size(); i++) {
        cout << (i+1) << ". " << compras[i] << endl;
    }

    // Operación 2: Eliminar por índice
    int elim;
    cout << "\nIngrese el número del producto que desea eliminar: ";
    cin >> elim;
    elim--;  // convertir a índice 0

    if(elim >= 0 && elim < static_cast<int>(compras.size())) {
        string eliminado = compras[elim];
        compras.erase(compras.begin() + elim);
        cout << "\nProducto eliminado: " << eliminado << endl;
    } else {
        cout << "Índice fuera de rango." << endl;
    }

    // Lista final
    cout << "\nLista final de compras:" << endl;
    for(size_t i = 0; i < compras.size(); i++) {
        cout << (i+1) << ". " << compras[i] << endl;
    }

    return 0;
}